cpp/src/parquet/column_reader.cc

```diff
@@ -486,9 +525,8 @@ std::shared_ptr<Page> SerializedPageReader::NextPage() {
           header.repetition_levels_byte_length, uncompressed_len, is_compressed,
           page_statistics);
     } else {
-      // We don't know what this page type is. We're allowed to skip non-data
-      // pages.
-      continue;
+      // We have already skipped non-data pages in ShouldSkipPage().
```

Comment:
nit: consider moving this detail into the exception, just in case there is a bug this will help people narrow it down quicker.

Reply:
Done